apollo-federation/src/connectors/json_selection/methods/public/ne.rs

```diff
@@ -0,0 +1,504 @@
+use apollo_compiler::collections::IndexMap;
+use serde_json_bytes::Value as JSON;
+use shape::Shape;
+use shape::location::SourceId;
+
+use crate::connectors::json_selection::ApplyToError;
+use crate::connectors::json_selection::ApplyToInternal;
+use crate::connectors::json_selection::MethodArgs;
+use crate::connectors::json_selection::VarsWithPathsMap;
+use crate::connectors::json_selection::immutable::InputPath;
+use crate::connectors::json_selection::location::Ranged;
+use crate::connectors::json_selection::location::WithRange;
+use crate::connectors::json_selection::methods::common::number_value_as_float;
+use crate::impl_arrow_method;
+
+impl_arrow_method!(NeMethod, ne_method, ne_shape);
+/// Returns true if argument is not equal to the applied to value or false if they are equal.
+/// Simple examples:
+///
+/// $(123)->ne(123)       results in false
+/// $(123)->ne(456)       results in true
+fn ne_method(
+    method_name: &WithRange<String>,
+    method_args: Option<&MethodArgs>,
+    data: &JSON,
+    vars: &VarsWithPathsMap,
+    input_path: &InputPath<JSON>,
+) -> (Option<JSON>, Vec<ApplyToError>) {
+    if let Some(MethodArgs { args, .. }) = method_args {
+        if let [arg] = args.as_slice() {
+            let (value_opt, arg_errors) = arg.apply_to_path(data, vars, input_path);
+            let mut apply_to_errors = arg_errors;
```

Comment:
```suggestion
            let (value_opt, mut apply_to_errors) = arg.apply_to_path(data, vars, input_path);
```

Not a suggestion, really, just an option. I don't feel strongly one way or the other.